src/Orckestra.Composer.Website/UI.Package/Typescript/Composer.ContentSearch/SearchResultsController.ts

```diff
@@ -33,21 +38,70 @@ module Orckestra.Composer {
                     TotalCount: Total,
                     SelectedSortBy,
                     AvailableSortBys,
+                    FacetsVisible: true,
                     Pagination: {
                         PagesCount: 1,
                         CurrentPage: 1,
                         PreviousPage: false,
                         NextPage: false,
                     },
+                    isOverriden: isOverriden,
                     isLoading: false
                 },
                 mounted() {
                     this.Pagination = this.getPagination(PagesCount)
                     self.eventHub.subscribe(ContentSearchEvents.SearchResultsLoaded, this.onSearchResultsLoaded);
+                    self.showFacetsService.getShowFacets().then(
+                        (value: boolean) => {
+                                this.FacetsVisible = value;
+                                if (!value) this.hideFacet(true);  // as an intial setup we hide the facet and ask for an update to be made  
+                           
+                        }, 
+                        (error: any) => {
+                            self.showFacetsService.setShowFacets(true);
+                        }
+                    );
+                    
                 },
                 computed: {
                 },
+                updated: function () {
+                    this.updateProductColumns();
+                },
                 methods: {
+                    hideFacet(update = false): void {
+                        document.getElementById("leftCol").classList.add("w-0-lg");
+                        document.getElementById("rightCol").classList.remove("col-lg-9");
+                        if(update) this.FacetsVisible = false; // setting this will trigger the "updated" function above only if requested
+                    },
+                    showFacet(): void {
+                        document.getElementById("leftCol").classList.remove("w-0-lg");
+                        document.getElementById("rightCol").classList.add("col-lg-9");
+                    },
+                    toggleFacet(): void {
+                        if (this.FacetsVisible) {
+                            this.hideFacet();
+                        }
+                        else {
+                            this.showFacet();
+                        }
+                        this.FacetsVisible = !this.FacetsVisible; // setting this will trigger the "updated" function above
+                        self.showFacetsService.setShowFacets(this.FacetsVisible);
+                    },
+                    updateProductColumns(){
+                        let searchContainer = document.getElementsByClassName("search-container");
```

Comment:
if we need some identifier for JS it is better to use id attribute for this, as class can be deleted
when you use id, you don'tt need to do this for loop
 for (let i=0; i < searchContainer.length; i++) {
 ...
 
 also is it next correct? Do nothing when isOverriden is true?
 Please rename here also this variable what it means, to understand better
 if (isOverriden) return;

Reply:
It cannot be an ID because it is applied to the product tile container (grid columns) and we don't always know the number of tiles there will be.

![Screen Shot 2022-11-29 at 7 14 55 AM](https://user-images.githubusercontent.com/44198946/204527484-be244a73-a687-4baa-8e75-22834fbb2bd7.png)

Reply:
Ok, then name is confusing - was thinking that this is a class for all search container, but this is just for tile
probably rename - product-col-container ?